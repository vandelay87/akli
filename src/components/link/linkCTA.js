import React from 'react'
import PropTypes from 'prop-types'
import { Link as GatsbyLink } from 'gatsby'
import styled from 'styled-components'
import { robotoRegular } from '../../styles/fonts'

const LinkCTA = ({ title, link, external }) => (
  <>
    {external ? (
      <StyledLink href={link} target="_blank" rel="noreferrer noopener">
        {title}
      </StyledLink>
    ) : (
      <StyledLink as={GatsbyLink} to={link}>
        {title}
      </StyledLink>
    )}
  </>
)

const StyledLink = styled('a')`
  ${robotoRegular}
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  letter-spacing: 0.03125em;
`

LinkCTA.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  external: PropTypes.bool,
}
LinkCTA.defaultProps = {
  external: false,
}

export default LinkCTA
