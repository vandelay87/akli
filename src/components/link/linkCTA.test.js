import renderer from 'react-test-renderer'
import { internal, external } from './linkCTA.stories'

describe('linkCTA.js', () => {
  it('renders component', () => {
    const internalTree = renderer.create(internal()).toJSON()
    const externalTree = renderer.create(external()).toJSON()

    expect(internalTree).toMatchSnapshot()
    expect(externalTree).toMatchSnapshot()
  })
})
