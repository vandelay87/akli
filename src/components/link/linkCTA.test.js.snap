// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkCTA.js renders component 1`] = `
<a
  aria-current="page"
  className="sc-bdVaJa inBooY"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={Object {}}
>
  click me
</a>
`;

exports[`linkCTA.js renders component 2`] = `
<a
  className="sc-bdVaJa inBooY"
  href="https://www.google.com"
  rel="noreferrer noopener"
  target="_blank"
>
  google.com
</a>
`;
